src/erlfmt_format.erl

```diff
@@ -344,57 +344,155 @@ flex_container(Meta, Values, Left, Right) ->
 call(Meta, Values, Left, Right) ->
     container_common(Meta, Values, Left, Right, break, last_fits).
 
-container_common(_Meta, [], Left, Right, _Combine, _Last) ->
+container_common(_Meta, [], Left, Right, _, _) ->
     concat(Left, Right);
-container_common(Meta, [Value | _] = Values, Left, Right, Combine, Last) ->
-    {HasTrailingComments, WasLastFits, ValuesD} =
-        container_exprs_to_algebra(Values, Last, []),
-    case HasTrailingComments orelse has_inner_break(Meta, Value) of
-        true ->
-            Doc = fold_doc(fun (D, Acc) -> line(concat(D, <<",">>), Acc) end, ValuesD),
-            surround(Left, <<"">>, concat(force_breaks(), Doc), <<"">>, Right);
-        false when Combine =:= break ->
-            Doc = fold_doc(fun (D, Acc) -> break(concat(D, <<",">>), Acc) end, ValuesD),
-            Wrapped = surround(Left, <<"">>, Doc, <<"">>, Right),
-            case WasLastFits of
-                true -> next_break_fits(Wrapped, disabled);
-                false -> Wrapped
-            end;
-        false when Combine =:= flex_break ->
-            Doc = fold_doc(
-                fun (D, Acc) -> flex_break(concat(D, <<",">>), Acc) end,
-                ValuesD
-            ),
-            Wrapped = group(concat(nest(concat(Left, Doc), ?INDENT, break), Right)),
-            case WasLastFits of
-                true -> next_break_fits(Wrapped, disabled);
-                false -> Wrapped
-            end
-    end.
+container_common(Meta, Values, Left, Right, Break, LastFits0) ->
+    SplitValues = init_last_and_comments(Values),
+    LastFits = last_fits(LastFits0, SplitValues),
+    HasLineBreak = has_line_break(Meta, SplitValues),
+    BreakFun = break_fun(Break, HasLineBreak),
```

Comment:
What do you think about having something like:
```erl
{BreakFun, SurroundFun} = break_and_surround_fun(Break, Meta, SplitValues)
```
It seems that they are both computed in a very similar way, but now it's split in two places.

This would allow avoiding passing `HasLineBreak` and `Break` arguments to `surround_container_values` in favour of single `SurroundFun` and moving `has_line_break` call inside of `break_and_surround_fun`.

Reply:
I can add SurroundFun, I agree that would look much better, but would prefer not to create a function that does two things `break_and_surround_fun`.  But I will also see what it looks like in practice.

Reply:
Lumping them together for me makes sense since they use exactly the same conditions. I agree the name is not that nice.

Reply:
I pushed the commit that doesn't lump them together, but also tried to lump them together, here is what the diff would be.

```
--- a/src/erlfmt_format.erl
+++ b/src/erlfmt_format.erl
@@ -349,12 +349,14 @@ container_common(_Meta, [], Left, Right, _, _) ->
 container_common(Meta, Values, Left, Right, Break, LastFits0) ->
     SplitValues = init_last_and_comments(Values),
     LastFits = last_fits(LastFits0, SplitValues),
-    HasLineBreak = has_line_break(Meta, SplitValues),
-    BreakFun = break_fun(Break, HasLineBreak),
-    SurroundFun = surround_fun(HasLineBreak, Break, LastFits),
+    {BreakFun, SurroundFun} = surround_and_break_fun(Meta, SplitValues, Break, LastFits),
     Doc = container_inner_values(SplitValues, BreakFun, LastFits),
     SurroundFun(Left, Doc, Right).
 
+surround_and_break_fun(Meta, SplitValues, Break, LastFits) ->
+    HasLineBreak = has_line_break(Meta, SplitValues),
+    {break_fun(Break, HasLineBreak), surround_fun(HasLineBreak, Break, LastFits)}.
+
```

Reply:
Needed to another type of HasBreak and now surround_and_break_fun started having more of a benefit, so I added it.